docs/reference/search/profile.asciidoc

```diff
@@ -916,6 +916,13 @@ The meaning of the stats are as follows:
 
     This is not currently used and will always report `0`. Currently aggregation profiling only times the shard level parts of the aggregation execution. Timing of the reduce phase will be added later.
 
+`build_leaf_collector`::
+    Time spent by the `getLeafCollector()` method collecting documents from
+    segments.
```

Comment:
This is mostly "time spent finishing up the last leaf and preparing for the next one". We do things like translate the last segment's ords into global ords and grab a reference to the reader for the next segment.